Modules/xSharePoint/DSCResources/MSFT_xSPWebAppPolicy/MSFT_xSPWebAppPolicy.psm1

```diff
@@ -56,63 +81,260 @@ function Set-TargetResource
     param
     (
         [parameter(Mandatory = $true)]  [System.String] $WebAppUrl,
-        [parameter(Mandatory = $true)]  [System.String] $UserName,
-        [parameter(Mandatory = $true)]  [ValidateSet("Deny All","Deny Write","Full Read", "Full Control")] [System.String] $PermissionLevel,
-        [parameter(Mandatory = $false)] [System.Boolean] $ActAsSystemUser,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $Members,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $MembersToInclude,
+        [parameter(Mandatory = $false)] [Microsoft.Management.Infrastructure.CimInstance[]] $MembersToExclude,
+        [parameter(Mandatory = $false)] [System.Boolean] $SetCacheAccountsPolicy,
         [parameter(Mandatory = $false)] [System.Management.Automation.PSCredential] $InstallAccount
     )
 
     Write-Verbose -Message "Setting web app policy for $UserName at $WebAppUrl"
 
-    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments $PSBoundParameters -ScriptBlock {
+    if ($Members -and (($MembersToInclude) -or ($MembersToExclude))) {
+        Throw "Cannot use the Members parameter together with the MembersToInclude or MembersToExclude parameters"
+    }
+
+    if (!$Members -and !$MembersToInclude -and !$MembersToExclude) {
+        throw "At least one of the following parameters must be specified: Members, MembersToInclude, MembersToExclude"
+    }
+
+    foreach ($member in $Members) {
+        if (($member.ActAsSystemAccount -eq $true) -and ($member.PermissionLevel -ne "Full Control")) {
+            throw "Members Parameter: You cannot specify ActAsSystemAccount with any other permission than Full Control"        
+        }
+    }
+
+    foreach ($member in $MembersToInclude) {
+        if (($member.ActAsSystemAccount -eq $true) -and ($member.PermissionLevel -ne "Full Control")) {
+            throw "MembersToInclude Parameter: You cannot specify ActAsSystemAccount with any other permission than Full Control"        
+        }
+    }
+    
+    $result = Invoke-xSharePointCommand -Credential $InstallAccount -Arguments @($PSBoundParameters,$PSScriptRoot) -ScriptBlock {
```

Comment:
$result is unused here (I'm cleaning this up as I go through some other resources as well) so can we take that off the front of this line?